tests/zfs-tests/include/libtest.shlib

```diff
@@ -3331,6 +3331,35 @@ function wait_replacing #pool
 }
 
 #
+# Wait for drives to enter a faulted state
+#
+# $1 pool name
+#
+function wait_faulted #pool
+{
+	typeset pool=${1:-$TESTPOOL}
+	while true; do
+		[[ "" != "$(zpool status $pool | grep 'FAULTED')" ]] && break
```

Comment:
this is more clearly written as:
```
zpool status $pool | grep -q FAULTED && break
```
please use grep's return codes for decisions rather than a (simple) parse of the output